### ISSUE TITLE
Question about core inversion recovery code

### ISSUE THREAD

**Issue by mathieuboudreau** [qMRLab/qMRLab#370]
Hey @ileppe and @agahkarakuzu !

Someone flagged a bug in Barral's original code in my fork (https://github.com/mathieuboudreau/Gold-Standard-Inversion-Recovery-T1-Mapping/issues/1). I was wondering if either of you encountered this while working on the qMRLab implementation, and what workaround did you have. It does appear that that variable contains all the important sequence information, per the similar simulation script that works for me: https://github.com/mathieuboudreau/Gold-Standard-Inversion-Recovery-T1-Mapping/blob/master/mfiles/mainSim.m

**Comment by agahkarakuzu** [comment]
It does not look like a bug to me.

Honestly, the information provided in the [issue](https://github.com/mathieuboudreau/Gold-Standard-Inversion-Recovery-T1-Mapping/issues/1) is not enough to understand the origin of the problem. We don't even know what is passed to the function as `extra`. 

See the following lines from the qMRLab script called for fitting, it works fine:

https://github.com/qMRLab/qMRLab/blob/699437e1fbbffde4ccd94f05aa6e3e5228dabb94/src/Models_Functions/IRfun/fitT1_IR.m#L35-L40

**Comment by mathieuboudreau** [comment]
By bug, I mean that the script calls a function, which that function calls a variable that was never defined or in scope (the sequence parameters). It just needs to be defined, but in it's current state Barral's code does not work with the example.

>We don't even know what is passed to the function as extra, where it was called from etc.

Yes we do, see this script for the simulations: https://github.com/mathieuboudreau/Gold-Standard-Inversion-Recovery-T1-Mapping/blob/master/mfiles/mainSim.m

All that is needed is for the mainScan.m function to do the same.

**Comment by agahkarakuzu** [comment]
What I meant was if the `mainScan.m` in your fork's master branch was run as is, which seemed unlikely because: 

There is a mismatch between the folder structure in the repo and what the script is trying to look for: 

https://github.com/mathieuboudreau/Gold-Standard-Inversion-Recovery-T1-Mapping/blob/230928a4e4e431e2b42fba0e801c56db38990aab/mfiles/mainScan.m#L15-L26

* It assumes that the script is called when the `pwd` is where `mainScan.m` located
* It looks for data in `data` folder, whereas the data is in `fitdata` folder 
* Filenames in `fitdata` are not `TestSingleSlice`, but `T1FitRD-NLS_TestSingleSlice.mat` etc

Only after dealing with these the following error (in the issue) can appear:

```
Undefined function or variable 'extra'.

Error in T1ScanExperiment (line 22)
nlsS = getNLSStruct(extra,1);

Error in mainScan (line 36)
T1ScanExperiment(loadStr, saveStr, method);
```

Which can be resolved by changing the following:

https://github.com/mathieuboudreau/Gold-Standard-Inversion-Recovery-T1-Mapping/blob/230928a4e4e431e2b42fba0e801c56db38990aab/mfiles/T1ScanExperiment.m#L22

with

```
%nlsS = getNLSStruct(extra,1);
extra = nlsS;
```

Then another error will be thrown because the code looks for the variable `data`, which is loaded as `ll_T1`. Similar variable name mismatch propagates into some other scripts such as `T1FitDisplayScan` because it also loads data from the same mat file... 

The code needs various fixes. How urgent is this? I can make a PR to your fork.

**Comment by mathieuboudreau** [comment]
>How urgent is this?

Not at all, I just wanted to open up a discussion here too since it might be related to the RRSG challenge. 

Thanks for your insights, I'll go through it deeper next week!

**Comment by agahkarakuzu** [comment]
> Thanks for your insights, I'll go through it deeper next week!

No problem 👍 The problems in the original repo have to do with variable naming, folder organization, assumptions, conditions etc peripheral factors. As long as we stick with qMRLab for RRSG challenge, we are isolated from all those issues as its own api for managing these.

**Comment by mathieuboudreau** [comment]
Yup - though I suspect at least a couple people might decide to try Barral's original code on their own anyways, so it'd be nice to be ready for those questions to when they arise.
